integration-tests/src/test/java/org/apache/fineract/integrationtests/LoanChargePaymentWithAdvancedPaymentAllocationTest.java

```diff
@@ -0,0 +1,325 @@
+        client = clientHelper.createClient(ClientHelper.defaultClientCreationRequest());
+        businessDateHelper = new BusinessDateHelper();
+        scheduleJobHelper = new SchedulerJobHelper(requestSpec);
+        savingsAccountHelper = new SavingsAccountHelper(requestSpec, responseSpec);
+        savingsProductHelper = new SavingsProductHelper();
+        commonLoanProductId = createLoanProduct("500", "15", "4", assetAccount, incomeAccount, expenseAccount, overpaymentAccount);
+    }
+
+    @Test
+    public void chargePaymentWithDefaultAllocationRuleTest() {
+        ;
+        final Account assetAccount = accountHelper.createAssetAccount();
+        final Account incomeAccount = accountHelper.createIncomeAccount();
+        final Account expenseAccount = accountHelper.createExpenseAccount();
+        final Account overpaymentAccount = accountHelper.createLiabilityAccount();
+
+        final String jobName = "Transfer Fee For Loans From Savings";
+        final String startDate = "10 April 2022";
+
+        GlobalConfigurationHelper.updateIsBusinessDateEnabled(requestSpec, responseSpec, Boolean.TRUE);
+        businessDateHelper.updateBusinessDate(new BusinessDateRequest().type(BusinessDateType.BUSINESS_DATE.getName()).date("2023.02.15")
+                .dateFormat("yyyy.MM.dd").locale("en"));
+
+        final Integer savingsId = createSavingsAccountDailyPosting(client.getClientId().intValue(), startDate);
+
+        savingsAccountHelper.depositToSavingsAccount(savingsId, "10000", startDate, CommonConstants.RESPONSE_RESOURCE_ID);
+
+        final PostLoansResponse loanResponse = applyForLoanApplication(client.getClientId(), commonLoanProductId, 1000L, 45, 15, 3, 0,
+                "01 January 2023", "01 January 2023");
+
+        int loanId = loanResponse.getLoanId().intValue();
+
+        loanTransactionHelper.updateLoan(loanId,
+                updateLoanJson(client.getClientId().intValue(), commonLoanProductId, savingsId.toString()));
+
+        loanTransactionHelper.approveLoan(loanResponse.getLoanId(), new PostLoansLoanIdRequest()
+                .approvedLoanAmount(BigDecimal.valueOf(1000)).dateFormat(DATETIME_PATTERN).approvedOnDate("01 January 2023").locale("en"));
+
+        loanTransactionHelper.disburseLoan(loanResponse.getLoanId(), new PostLoansLoanIdRequest().actualDisbursementDate("01 January 2023")
+                .dateFormat(DATETIME_PATTERN).transactionAmount(BigDecimal.valueOf(1000.00)).locale("en"));
+
+        final float feePortion = 50.0f;
+        final float penaltyPortion = 100.0f;
+
+        Integer flat = ChargesHelper.createCharges(requestSpec, responseSpec,
+                ChargesHelper.getLoanSpecifiedDueDateWithAccountTransferJSON(ChargesHelper.CHARGE_CALCULATION_TYPE_FLAT, "100", false));
+
+        LocalDate targetDate = LocalDate.of(2011, 3, 4);
+
+        String loanDetails = loanTransactionHelper.getLoanDetails(requestSpec, responseSpec, loanId);
```

Comment:
Please use the fineract-client to retrieve the loan details. This way you got the details in a type-safe way.